src/classes/DoubleLinkedList.js

```diff
@@ -0,0 +1,75 @@
+import { Node } from './Node';
+
+class DoubleLinkedList {
+  constructor(){
+    this.first = null;
+    this.last = null;
+    this.len = 0;
+  }
+
+  push(data) {
+    const node = new Node(data);
+
+    this.len += 1;
+    if (this.first === null) {
+      this.first = node;
+      this.last = node;
+    } else {
+      this.last.next = node;
+      node.prev = this.last;
+      this.last = node;
+    }
+  }
+
+  unshift(data) {
+    const node = new Node(data);
+
+    this.len += 1;
+    if (this.first === null) {
+      this.first = node;
+      this.last = node;
+    } else {
+      this.first.prev = node;
+      node.next = this.first;
+      this.first = node;
+    }
+  }
+
+  pop() {
+    if (!this.first) return;
+
+    const node = this.last;
+    this.last = this.last.prev;
+    this.last ? this.last.next = null : this.first = null;
+    this.len -= 1;
+
+    return node;
+  }
+
+  shift() {
+    if (!this.first) return;
+
+    const node = this.first;
+
+    this.first = this.first.next;
+    this.first ? this.first.prev = null : this.last = null;
+    this.len -= 1;
+
+    return node;
+  }
+
+  getValues() {
+    const resultList = [];
+    let node = this.first;
+
+    while (node) {
+      resultList.push(node.data);
+      node = node.next;
+    }
+
+    return resultList;
+  }
+
+}
```

Comment:
I think we should isolate the node we are returning

```suggestion
  }

  isolateNode(node) {
    node.prev = null;
    node.next = null;

    return node;
  }
}
```

Reply:
What do you mean by isolate?

Reply:
I'm thinking the .pop() method for example is doing something like this:

![Archivo](https://github.com/user-attachments/assets/bd1cd79c-588d-438f-af7f-04633cdf8140)


the returned node is still pointing to its prev. node, which can potentially give a badway access to the linkedlist